src/app/services/baw-api/account.service.spec.ts

```diff
@@ -8,214 +8,378 @@ import { Subject } from "rxjs";
+      | "apiUpdate"
+      | "apiDestroy",
+    url: string,
+    error: ApiErrorDetails
+  ) {
+    spyOn<any>(service as any, func).and.callFake((path: string) => {
+      expect(path).toBe(url);
+      const subject = new Subject();
+
+      setTimeout(() => {
+        subject.error(error);
+      }, 50);
+
+      return subject;
+    });
+  }
+
+  const shouldNotSucceed = () => {
+    fail("Service should not produce a data output");
+  };
+
+  const shouldNotFail = () => {
+    fail("Service should not produce an error");
+  };
+
   beforeEach(() => {
     TestBed.configureTestingModule({
       imports: [HttpClientTestingModule, RouterTestingModule],
       providers: [
         ...testAppInitializer,
-        { provide: BawApiService, useClass: MockBawApiService },
-        { provide: ApiCommon, useClass: MockApiCommon }
+        { provide: BawApiService, useClass: MockBawApiService }
       ]
     });
 
-    service = TestBed.get(UserService);
+    service = TestBed.get(AccountService);
     httpMock = TestBed.get(HttpTestingController);
   });
 
   afterEach(() => {
     httpMock.verify();
   });
 
-  it("should be created", () => {
-    expect(service).toBeTruthy();
-  });
+  describe("list", () => {
+    function createSuccess(path: string, model: User[]) {
+      spyOn(service as any, "apiList").and.callFake((_path: string) => {
+        expect(_path).toBe(path);
 
-  /**
-   * getUserAccount
-   */
-
-  it("should handle response", fakeAsync(() => {
-    spyOn<any>(service, "show").and.callFake(
-      (path: string, filters: Filters) => {
-        expect(path).toBe("/user_accounts/1");
-        expect(filters).toEqual(null);
-        const subject = new Subject<User>();
+        const subject = new Subject<User[]>();
 
         setTimeout(() => {
-          subject.next(
-            new User({
-              id: 1,
-              userName: "username",
-              rolesMask: 3,
-              rolesMaskNames: ["user"],
-              lastSeenAt: "1970-01-01T00:00:00.000"
-            })
-          );
+          subject.next(model);
           subject.complete();
         }, 50);
 
         return subject;
-      }
-    );
-
-    service.show(1).subscribe(
-      (user: User) => {
-        expect(user).toBeTruthy();
-        expect(user).toEqual(
-          new User({
-            id: 1,
-            userName: "username",
-            rolesMask: 3,
-            rolesMaskNames: ["user"],
-            lastSeenAt: "1970-01-01T00:00:00.000"
-          })
-        );
-      },
-      () => {
-        expect(true).toBeFalsy("Service should not return an error");
-      }
-    );
-
-    tick(100);
-  }));
-
-  it("should handle response with random id", fakeAsync(() => {
-    spyOn<any>(service, "show").and.callFake(
-      (path: string, filters: Filters) => {
-        expect(path).toBe("/user_accounts/5");
-        expect(filters).toEqual(null);
+      });
+    }
+
+    it("should handle empty response", fakeAsync(() => {
```

Comment:
why `fakeAsync`? why not just use the `async` keyword?

Reply:
I prefer `fakeAsync` because it has all of the functionality of `async` with additional control. `async` actually waits for the timer to finish, whereas `fakeAsync` allows me to kill the timer making tests run faster.